libvast/src/system/exporter.cpp

```diff
@@ -273,24 +293,35 @@ behavior exporter(stateful_actor<exporter_state>* self, expression expr,
         shutdown(self);
     },
     [=](extract_atom) {
+      auto& qs = self->state.query;
+      // Sanity check.
       VAST_DEBUG(self, "got request to extract all events");
-      if (self->state.query.requested == max_events) {
+      if (qs.requested == max_events) {
         VAST_WARNING(self, "ignores extract request, already getting all");
         return;
       }
-      self->state.query.requested = max_events;
+      // Configure state to get all remaining partition results.
+      qs.requested = max_events;
       ship_results(self);
       request_more_hits(self);
     },
-    [=](extract_atom, uint64_t requested) {
-      if (self->state.query.requested == max_events) {
+    [=](extract_atom, uint64_t client_demand) {
```

Comment:
I'm getting confused when I read `demand`, as it's a CAF term for flow control. What was wrong with `requested`? To me, it still signals best that this is "the number of results the user requested" for this extraction message.

Also, you're refactoring the local variable but the query statistics still have a `requested` variable.